roles/translator/src/main.rs

```diff
@@ -77,6 +81,131 @@ async fn main() {
+            State::UpstreamShutdown(err) => {
+                error!("SHUTDOWN from: {}", err);
+
+                // wait a random amount of time between 0 and 3000ms
+                // if all the downstreams try to reconnect at the same time, the upstream may fail
+                tokio::time::sleep(Duration::from_millis(1000)).await;
+                let mut rng = rand::thread_rng();
+                let wait_time = rng.gen_range(0..=3000);
+                tokio::time::sleep(Duration::from_millis(wait_time)).await;
+
+                // kill al the tasks
+                let task_collector_aborting = task_collector_.clone();
+                kill_tasks(task_collector_aborting.clone());
+
+                error!("Trying recconnecting to upstream");
+                start(
+                    rx_sv2_submit_shares_ext.clone(),
+                    tx_sv2_submit_shares_ext.clone(),
+                    tx_sv2_new_ext_mining_job.clone(),
+                    tx_sv2_set_new_prev_hash.clone(),
+                    tx_sv2_extranonce.clone(),
+                    rx_sv2_extranonce.clone(),
+                    rx_sv2_set_new_prev_hash.clone(),
+                    rx_sv2_new_ext_mining_job.clone(),
+                    rx_sv1_downstream.clone(),
+                    tx_sv1_bridge.clone(),
+                    tx_sv1_notify.clone(),
+                    target.clone(),
+                    tx_status.clone(),
+                    task_collector_.clone(),
+                )
+                .await;
+            }
+            State::Healthy(msg) => {
+                info!("HEALTHY message: {}", msg);
+            }
+        }
+    }
+}
+
+fn kill_tasks(task_collector: Arc<Mutex<Vec<(AbortHandle, String)>>>) {
+    let _ = task_collector.safe_lock(|t| {
+        while let Some(handle) = t.pop() {
+            handle.0.abort();
+            warn!("Killed task: {:?}", handle.1);
+        }
+    });
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn start<'a>(
+    rx_sv2_submit_shares_ext: Receiver<SubmitSharesExtended<'static>>,
+    tx_sv2_submit_shares_ext: Sender<SubmitSharesExtended<'static>>,
+    tx_sv2_new_ext_mining_job: Sender<NewExtendedMiningJob<'static>>,
+    tx_sv2_set_new_prev_hash: Sender<SetNewPrevHash<'static>>,
+    tx_sv2_extranonce: Sender<(ExtendedExtranonce, u32)>,
+    rx_sv2_extranonce: Receiver<(ExtendedExtranonce, u32)>,
+    rx_sv2_set_new_prev_hash: Receiver<SetNewPrevHash<'static>>,
+    rx_sv2_new_ext_mining_job: Receiver<NewExtendedMiningJob<'static>>,
+    rx_sv1_downstream: Receiver<DownstreamMessages>,
+    tx_sv1_bridge: Sender<DownstreamMessages>,
+    tx_sv1_notify: broadcast::Sender<server_to_client::Notify<'static>>,
+    target: Arc<Mutex<Vec<u8>>>,
+    tx_status: async_channel::Sender<Status<'static>>,
+    task_collector: Arc<Mutex<Vec<(AbortHandle, String)>>>,
+) {
+    let proxy_config = match process_cli_args() {
+        Ok(p) => p,
+        Err(e) => panic!("failed to load config: {}", e),
+    };
```

Comment:
Why do you need this here?
Where was it before your changes in this PR?

Reply:
I dont need that there. I changed it passing to the start function